mail_tracking/models/mail_tracking_email.py

```diff
@@ -0,0 +1,472 @@
+# Copyright 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import re
+import time
+import urllib.parse
+import uuid
+from datetime import datetime
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import AccessError
+from odoo.tools import email_split
+
+_logger = logging.getLogger(__name__)
+
+EVENT_OPEN_DELTA = 10  # seconds
+EVENT_CLICK_DELTA = 5  # seconds
+
+
+class MailTrackingEmail(models.Model):
+    _name = "mail.tracking.email"
+    _order = "time desc"
+    _rec_name = "display_name"
+    _description = "MailTracking email"
+
+    # This table is going to grow fast and to infinite, so we index:
+    # - name: Search in tree view
+    # - time: default order fields
+    # - recipient_address: Used for email_store calculation (non-store)
+    # - state: Search and group_by in tree view
+    name = fields.Char(string="Subject", readonly=True, index=True)
+    display_name = fields.Char(
+        readonly=True,
+        store=True,
+        compute="_compute_tracking_display_name",
+    )
+    timestamp = fields.Float(
+        string="UTC timestamp", readonly=True, digits="MailTracking Timestamp"
+    )
+    time = fields.Datetime(readonly=True, index=True)
+    date = fields.Date(readonly=True, compute="_compute_date", store=True)
+    mail_message_id = fields.Many2one(
+        string="Message", comodel_name="mail.message", readonly=True, index=True
```

Comment:
```suggestion
        string="Message (mail)", comodel_name="mail.message", readonly=True, index=True
```
Name conflicts with `message_id` field.